src/System.Runtime.Analyzers/Core/ProvideCorrectArgumentsToFormattingMethods.cs

```diff
@@ -124,11 +124,11 @@ private static int GetFormattingArguments(string format)
             // https://github.com/dotnet/coreclr/blob/bc146608854d1db9cdbcc0b08029a87754e12b49/src/mscorlib/src/System/Text/StringBuilder.cs#L1312
 
             // return count of this format - {index[,alignment][:formatString]}
-            var count = 0;
-
             var pos = 0;
             int len = format.Length;
 
+            var uniqueNumbers = new System.Collections.Generic.List<int>();
```

Comment:
Make use of a `Set` here, instead of having to check to see if the list already contains the number. Also, add a using for the type, don't specify the fully qualified path.